messaging/src/main/java/org/axonframework/queryhandling/DefaultQueryBusSpanFactory.java

```diff
@@ -0,0 +1,173 @@
+     *
+     * @param builder The builder to build the {@link DefaultQueryBusSpanFactory} from.
+     */
+    protected DefaultQueryBusSpanFactory(Builder builder) {
+        builder.validate();
+        this.spanFactory = builder.builderSpanFactory;
+        this.distributedInSameTrace = builder.distributedInSameTrace;
+    }
+
+
+    @Override
+    public Span createQuerySpan(QueryMessage<?, ?> queryMessage, boolean distributed) {
+        if (distributed) {
+            return spanFactory.createDispatchSpan(() -> "QueryBus.queryDistributed", queryMessage);
+        }
+        return spanFactory.createInternalSpan(() -> "QueryBus.query", queryMessage);
+    }
+
+    @Override
+    public Span createSubscriptionQuerySpan(SubscriptionQueryMessage<?, ?, ?> queryMessage, boolean distributed) {
+        if (distributed) {
+            return spanFactory.createDispatchSpan(() -> "QueryBus.subscriptionQueryDistributed", queryMessage);
+        }
+        return spanFactory.createInternalSpan(() -> "QueryBus.subscriptionQuery", queryMessage);
+    }
+
+    @Override
+    public Span createSubscriptionQueryProcessUpdateSpan(SubscriptionQueryUpdateMessage<?> updateMessage,
+                                                         SubscriptionQueryMessage<?, ?, ?> queryMessage) {
+        return spanFactory.createChildHandlerSpan(() -> "QueryBus.subscriptionQuery", updateMessage, queryMessage);
+    }
+
+    @Override
+    public Span createScatterGatherSpan(QueryMessage<?, ?> queryMessage, boolean distributed) {
+        if (distributed) {
+            return spanFactory.createDispatchSpan(() -> "QueryBus.scatterGatherQueryDistributed", queryMessage);
+        }
+        return spanFactory.createInternalSpan(() -> "QueryBus.scatterGatherQuery", queryMessage);
+    }
+
+    @Override
+    public Span createScatterGatherHandlerSpan(QueryMessage<?, ?> queryMessage, int handlerIndex) {
+        return spanFactory.createInternalSpan(() -> "QueryBus.scatterGatherQuery-" + handlerIndex, queryMessage);
+    }
+
+    @Override
+    public Span createStreamingQuerySpan(QueryMessage<?, ?> queryMessage, boolean distributed) {
+        if (distributed) {
+            return spanFactory.createDispatchSpan(() -> "QueryBus.streamingQueryDistributed", queryMessage);
+        }
+        return spanFactory.createInternalSpan(() -> "QueryBus.streamingQuery", queryMessage);
+    }
+
+    @Override
+    public Span createQueryProcessingSpan(QueryMessage<?, ?> queryMessage) {
+        if (distributedInSameTrace) {
+            return spanFactory.createChildHandlerSpan(() -> "QueryBus.processQueryMessage", queryMessage);
+        }
+        return spanFactory.createLinkedHandlerSpan(() -> "QueryBus.processQueryMessage", queryMessage);
+    }
+
+    @Override
+    public Span createResponseProcessingSpan(QueryMessage<?, ?> queryMessage) {
+        return spanFactory.createInternalSpan(() -> "QueryBus.processQueryResult", queryMessage);
```

Comment:
Shouldn't this span be called `QueryBus.processQueryResponse`?

Reply:
Fair enough, adjusted